test/com/google/javascript/jscomp/CollapsePropertiesTest.java

```diff
@@ -4203,4 +4209,51 @@ public void testTernaryExpression() {
         "var a = p ? x : {b: 1}; a.c = 2;", //
         "var a = p ? x : {b: 1}; var a$c = 2;");
   }
+
+  @Test
+  public void testModuleDynamicImport() {
```

Comment:
If we moved this test into the `/integration/` package, would we be able to avoid all the extra infrastructure around transpiling before unit tests to `CompilerTestCase.java`?

Reply:
It doesn't look like integration tests are setup to handle code splitting scenarios - none of the `test*` methods accept JSModule inputs and I still think we'd want to test this specific scenario in the Collapse Properties unit tests.

I'm all for avoiding transpilation in most tests, but module rewriting seems different as it can't really be disabled.

Reply:
OK, fair enough. I'm still hoping to avoid adding more to the CompilerTestCase API than necessary, and I have one suggestion for something that I think can be removed (`chunkOutputType`).